### PR TITLE
Document and extend OBSLocal

### PR DESCRIPTION
**Pull request by jreidinger**
- Documentation of OBSLocal extended and adapted to Python's standard format (reStructured Text).
- Added new methods to `OBSLocal.Project` to extend the current management of meta information
- Bonus: small fixes in `check_source_test.py`

### CODE REVIEW

File: tests/OBSLocal.py [hunk 1]
```diff
@@ -405,6 +422,18 @@ def remove(self):
 
 class Project(object):
     def __init__(self, name, reviewer={}, maintainer={}, project_links=[], create=True, with_repo=False):
+        """
+          Represents Project in OBS.
+
+              Parameters:
+                  name(str): Project name
+                  reviewer: TODO
+                  maintainerr: TODO
```
Comment (teclator) [inline tests/OBSLocal.py:431]:
```suggestion
                  maintainer: TODO
```

**Comment by ancorgs** [comment]
This doesn't seem to follow any of the four more extended docstring formats (see https://realpython.com/documenting-python-code/#docstring-formats, I have found the same list in several other places).

Other parts of this repository (like `osclib/stagingapi.py` or  `osclib/comments.py`) seem to be using reStructured Text. So I would suggest to stick to that instead of introducing our own (unofficial) variant.

File: tests/OBSLocal.py [hunk 2]
```diff
@@ -331,6 +381,17 @@ def create_project(self, name, reviewer={}, maintainer={}, project_links=[]):
         return self.projects[name]
 
     def submit_package(self, package=None, project=None):
+        """Creates submit request from package to target project.
+
+        Both have to exist, otherwise looks at create_submit_request method.
+
+        :param package: package to submit TODO: what means None here?
```
Comment (teclator) [inline tests/OBSLocal.py:388]: Looks like mandatory

Comment (mvidner) [reply tests/OBSLocal.py]: yes, I have committed a fix to that effect

**Review by mvidner — CHANGES_REQUESTED** [review]
Thanks @jreidinger and @ancorgs for a thorough documentation!

So we are documenting (and extending) some OBS API. Surely there is some other python binding of the same HTTP API in `osc`? We should link to it so that we don't reinvent the wheel now or in the future.

**Comment by mvidner** [comment]
... which is in https://github.com/openSUSE/osc/blob/master/osc/core.py

**Review by mvidner — APPROVED** [review]
Having added the links, approving :)

**Comment by mvidner** [comment]
@ancorgs excellent! thanks! still LGTM.

**Pull request merged**